### PR TITLE
Event System

### PR DESCRIPTION
**Pull request by beeperdeeper089**
**Addresses: None**

**Authors: @beeperdeeper089**

## Summary of changes
  - Builds in a robust event system (not hooked up yet do to window implementation currently in progress)

## Caveats
None.

## On approval
Merge.

### CODE REVIEW

**Comment by apachano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           

Clones added
============
- Sandbox/Source/Main.cpp  1
- Sandbox/Include/Sandbox/Sandbox.hpp  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GentenStudios/Phoenix/pullRequest?prid=4803288&bid=15983102)

**Comment by apachano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           

Clones added
============
- Sandbox/Include/Sandbox/Sandbox.hpp  1
- Sandbox/Source/Main.cpp  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GentenStudios/Phoenix/pullRequest?prid=4803288&bid=15983102)

**Comment by apachano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           

Clones added
============
- Sandbox/Include/Sandbox/Sandbox.hpp  1
- Sandbox/Source/Main.cpp  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GentenStudios/Phoenix/pullRequest?prid=4803288&bid=15983102)

**Comment by apachano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           

Clones added
============
- Sandbox/Source/Main.cpp  1
- Sandbox/Include/Sandbox/Sandbox.hpp  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GentenStudios/Phoenix/pullRequest?prid=4803288&bid=15983102)

**Comment by apachano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 12
           

Clones added
============
- Sandbox/Include/Sandbox/Sandbox.hpp  3
- Sandbox/Source/Main.cpp  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GentenStudios/Phoenix/pullRequest?prid=4803288&bid=15983102)

**Comment by beeperdeeper089** [comment]
that was the codacy thing

**Comment by apachano** [comment]
Idk what's up, its not registering that the recent PR went to develop. You may want to pull develop onto it but I will start trying to read over it

File: Engine/Source/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#add_subdirectory(Events)
```
Comment (apachano) [inline Engine/Source/CMakeLists.txt]: Should this be commented still?

Comment (beeperdeeper089) [reply Engine/Source/CMakeLists.txt]: wtf im so stupid, this shouldn't be here.

**Pull request merged**